scipy/signal/spectral.py

```diff
@@ -1679,7 +1679,7 @@ def detrend_func(d):
     if axis != -1:
         # Specify as positive axis index
         if axis < 0:
-            axis = len(result.shape)-1-axis
+            axis = len(result.shape)-1+axis
```

Comment:
do we even need the conditional `if axis != -1`? Seems like unnecessary branching